### PR TITLE
[Merged by Bors] - feat: Composition of the coercion `ℝ → ℂ` with algebraic operations

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Complex/Basic.lean [hunk 1]
```diff
@@ -628,6 +628,22 @@ def ofReal : ℝ →+* ℂ where
 theorem ofReal_eq_coe (r : ℝ) : ofReal r = r :=
   rfl
 
+variable {α : Type*}
+
+@[simp] lemma ofReal_comp_add (f g : α → ℝ) : ofReal' ∘ (f + g) = ofReal' ∘ f + ofReal' ∘ g :=
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Basic.lean:633]:
```suggestion
@[simp] lemma ofReal'_comp_add (f g : α → ℝ) : ofReal' ∘ (f + g) = ofReal' ∘ f + ofReal' ∘ g :=
```
etc

Comment (eric-wieser) [reply Mathlib/Data/Complex/Basic.lean:633]: Though I think `ofReal` is intended as an implementation detail?

Comment (YaelDillies) [reply Mathlib/Data/Complex/Basic.lean:633]: Your suggestion goes against the local naming convention. I am happy to change the local naming convention, but it's scope-creep for this PR

Comment (eric-wieser) [reply Mathlib/Data/Complex/Basic.lean:633]: Ah, I hadn't noticed that.

Maybe it would be better to write this with `(\u)`  notation?

Comment (YaelDillies) [reply Mathlib/Data/Complex/Basic.lean:633]: Yeah but then I need to type-ascript everything, which I thought wasn't really worth it when I can call the function directly

Comment (j-loreaux) [reply Mathlib/Data/Complex/Basic.lean:633]: Well, you should only need to do one side, but point taken nevertheless.

**Review by j-loreaux — COMMENTED** [review]
Any reason not to have the `comp_nsmul` and `comp_zsmul` variants?

bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**